astropy/convolution/tests/test_convolve.py

```diff
@@ -925,3 +927,32 @@ def test_non_square_kernel_asymmetric(boundary):
     image[6, 6] = 1
     result = convolve(image, kernel, normalize_kernel=False, boundary=boundary)
     assert_allclose(result[5:8, 4:9], kernel)
+
+
+@pytest.mark.parametrize(('boundary', 'normalize_kernel'),
+                         itertools.product(BOUNDARY_OPTIONS,
+                                           NORMALIZE_OPTIONS))
+def test_uninterpolated_nan_regions(boundary, normalize_kernel):
+    #8086
+    # Test NaN interpolation of contiguous NaN regions with kernels of size
+    # identical and greater than that of the region of NaN values.
+
+    # Test case: kernel.shape == NaN_region.shape
+    kernel = Gaussian2DKernel(1, 5, 5)
+    nan_centroid = np.full(kernel.shape, np.nan)
+    image = np.pad(nan_centroid, pad_width=kernel.shape[0]*2, mode='constant',
+                   constant_values=1)
+    try:
+        result = convolve(image, kernel, boundary=boundary, nan_treatment='interpolate',
+                          normalize_kernel=normalize_kernel)
+    except AstropyUserWarning:
+        pass
+    assert(np.isnan(result.sum()))
+
+    # Test case: kernel.shape > NaN_region.shape
+    nan_centroid = np.full((kernel.shape[0]-1, kernel.shape[1]-1), np.nan) # 1 smaller than kerenel
+    image = np.pad(nan_centroid, pad_width=kernel.shape[0]*2, mode='constant',
+                   constant_values=1)
+    result = convolve(image, kernel, boundary=boundary, nan_treatment='interpolate',
+                      normalize_kernel=normalize_kernel)
+    assert(~np.isnan(result.sum())) # Note: negation
```

Comment:
I prefer `~np.any(np.isnan(result))`.  It's slightly clearer, and it may(?) be slightly faster.  Probably not, since it's a two-pass process, though.... is there a `np.any_is_nan` method somewhere....?

Reply:
``np.isnan(result.sum())`` is faster than ``np.any(np.isnan(result))``

Reply:
But I'm still happy to change it as it is clearer and we shouldn't need to care about perf here.

Reply:
I'm fine either way.  I like the clarity of the slow version, but frankly, it's pretty obvious to me what this is doing at a glance.  This feels like such a common use case that I'm surprised there's no convenience function in numpy that does the fastest & clearest thing...

Reply:
Hmmmm, I wonder if it's because ``.sum()`` is implementation specific, i.e. it can be overloaded for the class type used and therefore can't be guaranteed to result in ``NaN``? Some aspect of this argument is a rationale for us not using, however, it would be ok in this test case. It is currently used in convolve, I should double check that it's ok.

Reply:
```
In [12]: %timeit np.isnan(x).all()
389 µs ± 4.59 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [13]: %timeit np.isnan(x).any()
419 µs ± 8.84 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [14]: %timeit np.isnan(x.sum())
377 µs ± 2.39 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Weirdly, I find that the location of the nan doesn't matter for `any` or `all`, which I would expect since they each only need to hit one nan/non-nan value to return.